### PR TITLE
chore: Promote some integration tests to js_unit_tests

### PR DESCRIPTION
**Pull request by mmastrac**
- Move a workers test to js_unit_tests and make it work
 - (slightly) repair the websocketstream_test and make it a JS unit test. This test was being ignored and rotted quite a bit, but there's some value in running as much of it as we can.
 - Merge the two websocket test files

### CODE REVIEW

File: cli/tests/integration/js_unit_tests.rs [hunk 1]
```diff
@@ -123,7 +124,7 @@ fn js_unit_test(test: String) {
     .arg("cli/tests/config/deno.json")
     .arg("--no-lock")
     .arg("--unstable")
-    .arg("--location=http://js-unit-tests/foo/bar")
+    .arg("--location=http://127.0.0.1:4545/")
```
Comment (mmastrac) [inline cli/tests/integration/js_unit_tests.rs:127]: Allows us to use the test webserver to fetch files.

File: cli/tests/unit/websocketstream_test.ts [hunk 2]
```diff
@@ -57,23 +65,25 @@ Deno.test("echo string tls", async () => {
...
-Deno.test("websocket error", async () => {
+Deno.test("websocket error", { sanitizeOps: false }, async () => {
   const ws = new WebSocketStream("wss://localhost:4242");
   await Promise.all([
+    // TODO(mmastrac): this exception should be tested
     assertRejects(
       () => ws.opened,
-      Deno.errors.UnexpectedEof,
-      "tls handshake eof",
+      // Deno.errors.UnexpectedEof,
```
Comment (mmastrac) [inline cli/tests/unit/websocketstream_test.ts:74]: Older style of assertRejects -- doesn't work in modern code.

File: cli/tests/unit/websocket_test.ts [hunk 3]
```diff
@@ -435,3 +435,304 @@ Deno.test(
     await server.finished;
   },
 );
+
+Deno.test("invalid scheme", () => {
```
Comment (mmastrac) [inline cli/tests/unit/websocket_test.ts:439]: Merged these tests -- we don't need to have two websocket tests.

File: cli/tests/unit/websocketstream_test.ts [hunk 4]
```diff
@@ -22,20 +21,28 @@ Deno.test("duplicate protocols", () => {
...
 
-Deno.test("connect & close custom valid code", async () => {
-  const ws = new WebSocketStream("ws://localhost:4242");
-  await ws.opened;
-  ws.close({ code: 1000 });
-  await ws.closed;
-});
+Deno.test(
+  "connect & close custom valid code",
+  { sanitizeOps: false },
```
Comment (mmastrac) [inline cli/tests/unit/websocketstream_test.ts:26]: Lots of sanitizer failures in this file. I'm pretty sure WebSocketStream is flaky code in general. Will file follow-up issues

**Review by dsherret — APPROVED** [review]
Nice!

**Pull request merged**